pandas/core/internals/blocks.py

```diff
@@ -1005,11 +1014,21 @@ def putmask(self, mask, new) -> list[Block]:
         new = extract_array(new, extract_numpy=True)
 
         if noop:
+            if using_cow:
+                return [self.copy(deep=False)]
             return [self]
```

Comment:
I think so, but it could also mess up something I am not thinking about right now. I‘d like to keep as is if you don’t feel strongly about it. Could our caching mechanism be a problem (only theoretically, because we should clear the cache when getting here I think), but there might be edge cases we are missing right now

Reply:
It's fine to leave as is for now. It will also be an easy clean-up once CoW is enforced.

Reply:
Yep exactly my thinking as well